src/coins.cpp

```diff
@@ -990,6 +990,11 @@ const CTxOut &CCoinsViewCache::GetOutputFor(const CTxIn& input) const
 }
 
 CAmount CCoinsViewCache::GetValueIn(const CTransaction& tx) const
+{
+    return GetTransparentValueIn(tx) + tx.GetShieldedValueIn();
+}
+
+CAmount CCoinsViewCache::GetTransparentValueIn(const CTransaction& tx) const
```

Comment:
I personally prefer the split here; it means that transparent logic is cleanly factored out into its own function rather than being commingled with shielded logic.